### PR TITLE
[UXIT-1479] Add Redirects and Refactor next.config.js

### PR DESCRIPTION
**Pull request by mirhamasala**
## 📝 Description

This PR includes several refactoring and enhancement changes aimed at improving the organization, maintainability, and functionality of the project. The key motivations include better modularization of the configuration, adding necessary redirects, and improving code readability. This refactor also addresses the need to include specific redirects for ecosystem projects and image assets, which were missing in the previous configuration.

- **Type:** Refactor / Enhancement

## 🛠️ Key Changes

- **Abstracted redirects into a separate module** to improve the organization of the Next.js configuration.
- **Included a redirect for the Ecosystem project "Muckrock"** to ensure proper routing for this resource.
- **Added image redirects** to route specific image assets to the root directory as required.
- **Refactored and modularized the Next.js configuration** for better readability and maintainability, grouping related configuration settings into separate constants.

## 📌 To-Do Before Merging

- [x] Review the new redirects to ensure all necessary routes are covered.
- [x] Verify that the refactored Next.js configuration works as expected in both development and production environments.

## 🧪 How to Test

- **Setup:** Ensure your development environment is up-to-date with the latest branch.
- **Steps to Test:**
  1. Run the application locally and navigate to the pages that are affected by the redirects to verify they work correctly. The new redirects are added under [64c2aa3](https://github.com/FilecoinFoundationWeb/filecoin-foundation/pull/609/commits/64c2aa35479e7532da47c4e082f640e52f0e3c3b) and [c8db400](https://github.com/FilecoinFoundationWeb/filecoin-foundation/pull/609/commits/c8db4006054b8842a87d74bb79521de376f24861).
  2. Check that image URLs are properly redirected to the root as specified.
  3. Review the Next.js configuration to ensure that it remains functional and that all webpack rules are applied correctly.
- **Expected Results:** Redirects should work seamlessly, and the application should behave as expected without errors.

## 🔖 Resources

- [Next.js Redirects Documentation](https://nextjs.org/docs/api-reference/next.config.js/redirects)
- [Webpack Configuration Documentation](https://webpack.js.org/configuration/)

### CODE REVIEW

File: redirects.js [hunk 1]
```diff
@@ -149,6 +149,53 @@ const redirects = [
...
+  {
+    source: '/assets/svg/bg-index.svg',
+    destination: '/',
+    permanent: true,
+  },
+  {
+    source: '/assets/svg/logo.svg',
+    destination: '/assets/images/logo.svg',
+    permanent: true,
+  },
```
Comment (barbaraperic) [inline redirects.js:241]: `destination: "/"`?

File: redirects.js [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+  {
+    source: '/assets/svg/bg-index.svg',
+    destination: '/',
+    permanent: true,
+  },
+  {
+    source: '/assets/svg/logo.svg',
+    destination: '/assets/images/logo.svg',
```
Comment (barbaraperic) [inline redirects.js:239]:
```suggestion
    destination: '/',
```

**Comment by barbaraperic** [comment]
Looks good, love the organisation ✨ One small type in `redirect.js` - wondering if we want to use constants for frequently used destinations to avoid typos, like for "/", and some other that repeat a lot, like "/about"

File: redirects.js (hunk 2 above)
Comment (mirhamasala) [reply redirects.js:239]: Actually, this one is correct.

**Comment by mirhamasala** [comment]
> Wondering if we want to use constants for frequently used destinations to avoid typos, like for "/", and some other that repeat a lot, like "/about"

Yes - I would like that. I meant to try it out but forgot in the end. Let me see now. 😵‍💫

**Pull request merged**